src/nvim/vterm/encoding.c

```diff
@@ -164,115 +163,7 @@ static VTermEncoding encoding_utf8 = {
-                           const char bytes[], size_t *pos, size_t bytelen)
-{
-  int is_gr = bytes[*pos] & 0x80;
-
-  for (; *pos < bytelen && *cpi < cplen; (*pos)++) {
-    uint8_t c = (uint8_t)(bytes[*pos] ^ is_gr);
-
-    if (c < 0x20 || c == 0x7f || c >= 0x80) {
-      return;
-    }
-
-    cp[(*cpi)++] = c;
-  }
-}
-
-static VTermEncoding encoding_usascii = {
-  .decode = &decode_usascii,
-};
-
-struct StaticTableEncoding {
-  const VTermEncoding enc;
-  const uint32_t chars[128];
-};
-
-static void decode_table(VTermEncoding *enc, void *data, uint32_t cp[], int *cpi, int cplen,
-                         const char bytes[], size_t *pos, size_t bytelen)
+VTermEncoding *vterm_lookup_encoding(void)
 {
-  struct StaticTableEncoding *table = (struct StaticTableEncoding *)enc;
-  int is_gr = bytes[*pos] & 0x80;
-
-  for (; *pos < bytelen && *cpi < cplen; (*pos)++) {
-    uint8_t c = (uint8_t)(bytes[*pos] ^ is_gr);
-
-    if (c < 0x20 || c == 0x7f || c >= 0x80) {
-      return;
-    }
-
-    if (table->chars[c]) {
-      cp[(*cpi)++] = table->chars[c];
-    } else {
-      cp[(*cpi)++] = c;
-    }
-  }
-}
-
-static const struct StaticTableEncoding encoding_DECdrawing = {
-  { .decode = &decode_table },
-  {
-    [0x60] = 0x25C6,  // BLACK DIAMOND
-    [0x61] = 0x2592,  // MEDIUM SHADE (checkerboard)
-    [0x62] = 0x2409,  // SYMBOL FOR HORIZONTAL TAB
-    [0x63] = 0x240C,  // SYMBOL FOR FORM FEED
-    [0x64] = 0x240D,  // SYMBOL FOR CARRIAGE RETURN
-    [0x65] = 0x240A,  // SYMBOL FOR LINE FEED
-    [0x66] = 0x00B0,  // DEGREE SIGN
-    [0x67] = 0x00B1,  // PLUS-MINUS SIGN (plus or minus)
-    [0x68] = 0x2424,  // SYMBOL FOR NEW LINE
-    [0x69] = 0x240B,  // SYMBOL FOR VERTICAL TAB
-    [0x6a] = 0x2518,  // BOX DRAWINGS LIGHT UP AND LEFT (bottom-right corner)
-    [0x6b] = 0x2510,  // BOX DRAWINGS LIGHT DOWN AND LEFT (top-right corner)
-    [0x6c] = 0x250C,  // BOX DRAWINGS LIGHT DOWN AND RIGHT (top-left corner)
-    [0x6d] = 0x2514,  // BOX DRAWINGS LIGHT UP AND RIGHT (bottom-left corner)
-    [0x6e] = 0x253C,  // BOX DRAWINGS LIGHT VERTICAL AND HORIZONTAL (crossing lines)
-    [0x6f] = 0x23BA,  // HORIZONTAL SCAN LINE-1
-    [0x70] = 0x23BB,  // HORIZONTAL SCAN LINE-3
-    [0x71] = 0x2500,  // BOX DRAWINGS LIGHT HORIZONTAL
-    [0x72] = 0x23BC,  // HORIZONTAL SCAN LINE-7
-    [0x73] = 0x23BD,  // HORIZONTAL SCAN LINE-9
-    [0x74] = 0x251C,  // BOX DRAWINGS LIGHT VERTICAL AND RIGHT
-    [0x75] = 0x2524,  // BOX DRAWINGS LIGHT VERTICAL AND LEFT
-    [0x76] = 0x2534,  // BOX DRAWINGS LIGHT UP AND HORIZONTAL
-    [0x77] = 0x252C,  // BOX DRAWINGS LIGHT DOWN AND HORIZONTAL
-    [0x78] = 0x2502,  // BOX DRAWINGS LIGHT VERTICAL
-    [0x79] = 0x2A7D,  // LESS-THAN OR SLANTED EQUAL-TO
-    [0x7a] = 0x2A7E,  // GREATER-THAN OR SLANTED EQUAL-TO
-    [0x7b] = 0x03C0,  // GREEK SMALL LETTER PI
-    [0x7c] = 0x2260,  // NOT EQUAL TO
-    [0x7d] = 0x00A3,  // POUND SIGN
-    [0x7e] = 0x00B7,  // MIDDLE DOT
-  }
-};
```

Comment:
I think we should keep the line drawing character set since there is at least one known application which uses it (and box drawing is not an extremely uncommon use case).

foot, Ghostty, and [libvaxis](https://github.com/rockorager/libvaxis) all support the box drawing character set (possibly others, these are just the ones I know of for sure).

Reply:
But are they using _these_ characters and not the unicode ones?

(Link from chat for the "one known application": https://github.com/pimutils/khal)

Reply:
Actually it's the [vaxis](https://git.sr.ht/~rockorager/vaxis) embedded terminal that supports that character set. It didn't initially but we received bug reports in `aerc` for `khal` which used that charset for the box drawing glyphs.

Reply:
OK, just wanted to make sure. And they don't prefer the unicode set if that is available?

Reply:
And for the record, we're talking about  https://en.wikipedia.org/wiki/DEC_Special_Graphics

Reply:
These work by mapping ascii codepoints to other codepoints. So switching the charset to `0`, you can print a `j` and get a `┘`

Reply:
Just like `#` to `£` 🫖

Reply:
Also sorry, I told Greg the wrong application. It was [calcurse](https://calcurse.org/) which uses the box drawing charset

Reply:
> At least [Ghostty is](https://github.com/ghostty-org/ghostty/blob/799f5b823904a2076a49c494ae50b486ac519368/src/terminal/charsets.zig#L54-L84).

[ghostty implements the british and us encoding as well though](https://github.com/ghostty-org/ghostty/blob/799f5b823904a2076a49c494ae50b486ac519368/src/terminal/charsets.zig#L17-L20)

Reply:
> It didn't initially but we received bug reports in aerc for khal which used that charset for the box drawing glyphs.

@rockorager do you have a link to it so I can get more context?

Reply:
OK, the drawing character set is widely supported by most terminals and tmux, so it makes sense to keep it.